packages/react-native-windows-init/src/Cli.ts

```diff
@@ -0,0 +1,291 @@
+        pkg,
+        (err: any, details: {[key: string]: object}) => {
+          if (err) {
+            reject(err);
+          } else if (details) {
+            const versions = Object.keys(details);
+            if (versions.length > 0) {
+              const candidates = versions
+                .filter(_ => semver.satisfies(_, versionSemVer))
+                .sort(semver.rcompare);
+              if (candidates && candidates.length > 0) {
+                resolve(candidates[0]);
+                return;
+              }
+            }
+          }
+          reject(
+            new Error(`No matching version of ${pkg}@${versionSemVer} found`),
+          );
+        },
+      );
+    } else {
+      // Assume that versionSemVer is actually a tag
+      npm.packages.release(
+        pkg,
+        versionSemVer,
+        (err: any, details: {version: string}[]) => {
+          if (err) {
+            reject(err);
+          } else if (details && details.length > 0) {
+            resolve(details[0].version);
+            return;
+          }
+          reject(
+            new Error(`No matching version of ${pkg}@${versionSemVer} found`),
+          );
+        },
+      );
+    }
+  });
+}
+
+async function getLatestMatchingRNWVersion(
+  versionSemVer: string,
+): Promise<string> {
+  try {
+    const version = await getLatestMatchingVersion(
+      'react-native-windows',
+      versionSemVer,
+    );
+    if (!version) {
+      console.error(
+        `Error: No version of react-native-windows@${versionSemVer} found`,
+      );
+      process.exit(EXITCODE_NO_MATCHING_RNW);
+    }
+    return version;
+  } catch (err) {
+    console.error(
+      `Error: No version of react-native-windows@${versionSemVer} found`,
+    );
+    process.exit(EXITCODE_NO_MATCHING_RNW);
+  }
+}
+
+/**
+ * Check if project is using Yarn (has `yarn.lock` in the tree)
+ */
+function isProjectUsingYarn(cwd: string) {
+  return findUp.sync('yarn.lock', {cwd});
+}
+
+async function doMain() {
+  try {
+    const name = getReactNativeAppName();
+    const ns = argv.namespace || name;
+    let version = argv.version;
+
+    if (!version) {
+      const rnVersion = getReactNativeVersion();
+      version = getDefaultReactNativeWindowsSemVerForReactNativeVersion(
+        rnVersion,
+      );
+    }
+
+    let rnwResolvedVersion = await getLatestMatchingRNWVersion(version);
+
+    if (!argv.version) {
+      console.log(
+        `Latest matching version of ${chalk.bold(
+          'react-native-windows',
+        )} for ${chalk.green('react-native')}@${chalk.cyan(
+          getReactNativeVersion(),
+        )} is ${chalk.green('react-native-windows')}@${chalk.cyan(
+          rnwResolvedVersion,
+        )}`,
+      );
+
+      if (semver.prerelease(rnwResolvedVersion)) {
+        let rnwLatestVersion = await getLatestMatchingRNWVersion('latest');
+        console.warn(
+          `
+${chalk.green('react-native-windows')}@${chalk.cyan(
```

Comment:
Wow. eslint really mutilated this. Might be more readable to temporarily disable eslint here if we can